app/src/main/java/org/oppia/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -555,56 +556,30 @@ class StatePlayerRecyclerViewAssembler private constructor(
 
     /** Adds support for displaying state content to the learner. */
     fun addContentSupport(): Builder {
-      if (fragment is QuestionPlayerFragment)
-        adapterBuilder.registerViewBinder(
-          viewType = StateItemViewModel.ViewType.CONTENT,
-          inflateView = { parent ->
-            QuestionPlayerContentItemBinding.inflate(
-              LayoutInflater.from(parent.context),
-              parent,
-              /* attachToParent= */ false
-            ).root
-          },
-          bindView = { view, viewModel ->
-            val binding = DataBindingUtil.findBinding<QuestionPlayerContentItemBinding>(view)!!
-            val contentViewModel = viewModel as ContentViewModel
-            binding.htmlContent =
-              htmlParserFactory.create(
-                resourceBucketName,
-                entityType,
-                contentViewModel.gcsEntityId,
-                imageCenterAlign = true
-              ).parseOppiaHtml(
-                contentViewModel.htmlContent.toString(),
-                binding.questionPlayerContentTextView
-              )
-          }
-        )
-      else {
-        adapterBuilder.registerViewBinder(
-          viewType = StateItemViewModel.ViewType.CONTENT,
-          inflateView = { parent ->
-            ContentItemBinding.inflate(
-              LayoutInflater.from(parent.context),
-              parent,
-              /* attachToParent= */ false
-            ).root
-          },
-          bindView = { view, viewModel ->
-            val binding = DataBindingUtil.findBinding<ContentItemBinding>(view)!!
-            val contentViewModel = viewModel as ContentViewModel
-            binding.htmlContent =
-              htmlParserFactory.create(
-                resourceBucketName,
-                entityType,
-                contentViewModel.gcsEntityId,
-                imageCenterAlign = true
-              ).parseOppiaHtml(
-                contentViewModel.htmlContent.toString(), binding.contentTextView
-              )
-          }
-        )
-      }
+      adapterBuilder.registerViewBinder(
+        viewType = StateItemViewModel.ViewType.CONTENT,
+        inflateView = { parent ->
+          ContentItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            /* attachToParent= */ false
+          ).root
+        },
+        bindView = { view, viewModel ->
+          val binding = DataBindingUtil.findBinding<ContentItemBinding>(view)!!
+          val contentViewModel = viewModel as ContentViewModel
+          binding.viewModel = explorationContentViewModel
+          binding.htmlContent =
+            htmlParserFactory.create(
+              resourceBucketName,
+              entityType,
+              contentViewModel.gcsEntityId,
+              imageCenterAlign = true
+            ).parseOppiaHtml(
+              contentViewModel.htmlContent.toString(), binding.contentTextView
+            )
+        }
+      )
```

Comment:
Here, now `viewModel` and `contentViewModel` are not getting used directly and also the `explorationContentViewModel` is actually not a viewmodel which we are setting in xml file.